### PR TITLE
microrl: add configuration option for prompt coloring

### PR DESCRIPTION
**Pull request by iperry**
Add a configuration option for disabling output of ANSI color escape codes. This option is defaulted off to match existing behavior.

### CODE REVIEW

File: src/microrl/microrl.c [hunk 1]
```diff
@@ -244,9 +244,13 @@ MICRORL_CFG_STATIC_INLINE void prv_cmdline_buf_reset(microrl_t* mrl) {
  * \param[in]       mrl: \ref microrl_t working instance
  */
 MICRORL_CFG_STATIC_INLINE void prv_terminal_print_prompt(microrl_t* mrl) {
+#if MICRORL_CFG_USE_PROMPT_COLOR
     mrl->out_fn(mrl, MICRORL_CFG_PROMPT_COLOR);
+#endif
     mrl->out_fn(mrl, mrl->prompt_ptr);
+#if MICRORL_CFG_USE_PROMPT_COLOR
     mrl->out_fn(mrl, MICRORL_COLOR_DEFAULT);
+#endif
```
Comment (dimmykar) [inline src/microrl/microrl.c:253]:
```suggestion
#if MICRORL_CFG_USE_PROMPT_COLOR
    mrl->out_fn(mrl, MICRORL_CFG_PROMPT_COLOR);
    mrl->out_fn(mrl, mrl->prompt_ptr);
    mrl->out_fn(mrl, MICRORL_COLOR_DEFAULT);
#else
    mrl->out_fn(mrl, mrl->prompt_ptr);
#endif
```

File: src/include/microrl/microrl_config.h [hunk 2]
```diff
@@ -74,6 +74,13 @@ extern "C" {
 #define MICRORL_CFG_PROMPT_STRING             "> "
 #endif
 
+/**
+ * \brief           Enable ANSI color escape sequences
```
Comment (dimmykar) [inline src/include/microrl/microrl_config.h:78]:
```suggestion
 * \brief           Enable ANSI color escape sequences in prompt
```

**Review by dimmykar — CHANGES_REQUESTED** [review]
Hello!) Thank you for your contribution to the project.

Yes, this config can really be useful for some developers. I would like to ask you to

1. Squash your commit with the changes I suggested
2. Add note to the changelog about the new config

After that I will merge your request and release new version with this feature

**Comment by iperry** [comment]
> Hello!) Thank you for your contribution to the project.
> 
> Yes, this config can really be useful for some developers. I would like to ask you to
> 
> 1. Squash your commit with the changes I suggested
> 2. Add note to the changelog about the new config
> 
> After that I will merge your request and release new version with this feature

Thanks. I think I've made the requested modifications, let me know if not.

**Review by dimmykar — APPROVED** [review]
Now all looks good!) Thanks for your contribution!

I will publish `v2.5.0` release in a while

**Pull request merged**